plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -422,9 +452,30 @@ BEGIN
 	EXEC sp_executesql @SqlStatement
 
 END
-ELSE
+ELSE IF @EngineEdition IN (2,3,4) /*Standard,Enterprise,Express*/
 BEGIN
 
+	DECLARE @MajorMinorVersion AS int = CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') as nvarchar),4) AS int) * 100 + CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') as nvarchar),3) AS int)
+
+	DECLARE @Columns as nvarchar(max) = ''
+	DECLARE @Tables as nvarchar(max) = ''
+
+	IF @MajorMinorVersion >= 1050 BEGIN 
+		/* [volume_mount_point] TRIMS trailing "\" which are not allowed in InfluxDB */
+		SET @Columns += N',LEFT(vs.[volume_mount_point], LEN(vs.[volume_mount_point])-(PATINDEX(''%[^\]%'',REVERSE([volume_mount_point]))-1)) AS [volume_mount_point]'
```

Comment:
Let's do something for #7558 first so we can avoid it in the query.

Reply:
This code removes the trailing ```\``` from the string. It looks complex but it's the safest way to remove all the trailing chars in case the strings ends with multiple ```\```. (which does not make any sense but it's better to be safe)

Reply:
I hate for you to have to do this in query, and for outputs that do support trailing slashes it imposes the line protocol limitations on them too.  I opened #7652, let's see what others think of having the solution there.

Reply:
Since this has been fixed on the telegraf side I will try it out and then make a PR to avoid this logic in the query itself